front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -4,154 +4,124 @@ import { keyBy } from 'lodash';
+  pathProperties?: PathPropertiesFormatted,
+  selectedTrainSchedule?: TrainScheduleResult,
   path?: PathfindingResultSuccess
-): TimeStopsRow[] {
+): TimeStopsRow[] => {
   const { t } = useTranslation('timesStops');
 
-  const scheduleByAt: Record<string, ScheduleEntry> = keyBy(selectedTrainSchedule.schedule, 'at');
-  const suggestedOperationalPoints: SuggestedOP[] = useMemo(
-    () =>
-      path?.length
-        ? formatSuggestedOperationalPoints(
-            pathProperties.operationalPoints,
-            pathProperties.geometry,
-            path.length
-          )
-        : [],
-    [pathProperties.operationalPoints, pathProperties.geometry]
-  );
-
-  const pathStepsWithPositionOnPath = useMemo(() => {
-    if (!path || !selectedTrainSchedule) return [];
-    return selectedTrainSchedule.path.map((pathStep, index) => ({
-      ...pathStep,
-      positionOnPath: path.path_item_positions[index],
-    }));
-  }, [path, selectedTrainSchedule]);
-
-  const pathStepsWithOpPointIndices = useMemo(
-    () =>
-      selectedTrainSchedule.path
-        .filter((pathStep): pathStep is TrainScheduleBasePathWithUic => 'uic' in pathStep)
-        .map((pathStep) => {
-          const correspondingOpPointIndex = suggestedOperationalPoints.findIndex(
-            (sugOpPoint) =>
-              'uic' in pathStep &&
-              sugOpPoint.uic === pathStep.uic &&
-              sugOpPoint.ch === pathStep.secondary_code
-          );
-          return {
-            ...pathStep,
-            correspondingOpPointIndex,
-          };
-        }),
-    [selectedTrainSchedule, suggestedOperationalPoints]
-  );
-  const pathStepsByUic = keyBy(
-    pathStepsWithOpPointIndices,
-    (pathStep) => `${pathStep.uic}-${pathStep.secondary_code}`
-  );
-
-  const outputTableData = useMemo(() => {
-    const pathStepRows = pathStepsWithPositionOnPath.map((pathStep) => {
-      const schedule = scheduleByAt[pathStep.id];
-      const scheduleData = computeScheduleData(schedule);
-      return { ...pathStep, ...formatScheduleData(scheduleData) };
-    });
+  const scheduleByAt: Record<string, ScheduleEntry> = keyBy(selectedTrainSchedule?.schedule, 'at');
 
-    const suggestedOpRows = suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
-      const opPoint = operationalPoints.find((op) => op.id === sugOpPoint.opId);
-      if (!opPoint) {
-        return sugOpPoint;
-      }
-      const nextOpPoint = findNextScheduledOpPoint(
-        operationalPoints,
-        pathStepsWithOpPointIndices,
-        sugOpIndex
+  const pathStepRows = useMemo(() => {
+    if (!path || !selectedTrainSchedule || !trainSummary?.pathItemTimes) return [];
+
+    const startDatetime = new Date(selectedTrainSchedule.start_time);
+
+    const result: (TimeStopsRow & { positionOnPath: number })[] = [];
+    let previousTime = new Date(selectedTrainSchedule.start_time);
+
+    for (let index = 0; index < selectedTrainSchedule.path.length; index += 1) {
```

Comment:
I replaced the for loop by a map, but I had to keep the previous time, because I don't have the arrival or departure in the `path[index-1]` since it's only `pathStep` and not `pathStepRow` :/

I think it's cleaner to keep the `previousTime`, if it's ok for you

Reply:
Yes works for me thank you :)